metoffice_ec2/subset.py

```diff
@@ -12,14 +12,18 @@
 
 def subset(
     dataset: xr.Dataset,
-    height_meters: Optional[List[int]] = None,
+    height_meters: Optional[List[float]] = None,
     north: Optional[float] = None,
     east: Optional[float] = None,
     south: Optional[float] = None,
     west: Optional[float] = None,
 ) -> xr.Dataset:
 
-    if height_meters is not None:
+    if (
+        height_meters is not None
+        and not isinstance(height_meters, float)
```

Comment:
According to the type annotations in the function signature, `height_meters` should never be a float (although, of course, the Python interpreter doesn't actually care about the type annotations!).  So maybe the function should fail noisily if it `height_meters` is a float.  e.g. the function could start with:

```python
if height_meters is not None and not isinstance(height_meters, list):
    raise ValueError('height_meters must be a list or or None.')
```

Reply:
yeah, I thought so too, but one of the NetCDF files I pulled from the Met Office actually has height_meters as a float. Not sure where it's coming from, can investigate it more, but without this line, the script crashes.

Reply:
OK.  Good point.  In which case, maybe we should change the function signature to indicate that `height_meters` can be either a `float` or a `List[float]`.  You can do that with something like this in the function signature:

```python
height_meters: Optional[Union[float, List[float]]] = None
```

But, when this function receives `height_meters` as a `float` (as distinct from a `List[float]`) then this function must do something with `height_meters` and not silently ignore `height_meters`.

Reply:
Ah really? I thought it would just select the right data if it gets height_meters.
If I remember correctly then the 1D examples I saw where all flat, and not inside an array. Didn't seem like you need to select anything in that case.

Will add the type annotation.